### ISSUE TITLE
Results of directly running `model_segnet_single.py` 

### ISSUE THREAD

**Issue by LcDog** [lorenmt/mtan#14]
Hi @lorenmt , thanks for releasing your code!

I am running the single task script `model_segnet_single.py` and the final result seems to be not the same as reported in your paper. Here is my training log [semantic-1022070532.log](https://github.com/lorenmt/mtan/files/3756387/semantic-1022070532.log).

I just ran the script without modifying any params setting in the code. The best is about 12%, while in the paper the best is 15%. Could you please check the params setting?

Thanks.

**Comment by lorenmt** [comment]
Hello,

Thanks for your report. I checked the log you attached. It has clearly something wrong during the training stage, as the loss stuck around 1.74. 

As I explained in the 'readme', the final result should eventually slightly better than the number reported in the paper due to some additional effort to further optimize the code. So it's suggested to re-run the code by yourself in order to have a fair comparison to your proposed method.

I just reran the code as you suggest `model_segnet_single.py` without any modification to the parameter. (What I only did is to move scheduler.step() to the last as to be consistent for the newer version of pytorch) 

This is what I have for the first 8 epochs: 
![image](https://user-images.githubusercontent.com/6267679/67301387-3a0c0600-f4e7-11e9-9e7d-a6e91431e749.png)

Which is significantly different from the log you provide.

Could you provide further information so I may help? Or you can take a deeper look of your dataset management or other possible modifications?

Thanks.

**Comment by LcDog** [comment]
Thanks for your quick reply!

I am using the original code (without modification) and the pre-processed NYUv2 dataset provided in README. I have no idea why our results are so different. I directly ran the code again and got
```
Parameter Space: ABS: 24981069.0, REL: 1.0000

LOSS FORMAT: SEMANTIC_LOSS MEAN_IOU PIX_ACC
DEPTH_LOSS ABS_ERR REL_ERR
NORMAL_LOSS MEAN MED <11.25 <22.5 <30

Epoch: 0000 | TRAIN: 1.9737 0.0786 0.3567 TEST: 1.8311 0.1100 0.3978
Epoch: 0001 | TRAIN: 1.7701 0.1055 0.4153 TEST: 1.7933 0.1118 0.4045
Epoch: 0002 | TRAIN: 1.7575 0.1069 0.4179 TEST: 1.7888 0.1112 0.4076
Epoch: 0003 | TRAIN: 1.7567 0.1067 0.4185 TEST: 1.8055 0.1076 0.4031
Epoch: 0004 | TRAIN: 1.7520 0.1080 0.4211 TEST: 1.7995 0.1084 0.4043
Epoch: 0005 | TRAIN: 1.7563 0.1080 0.4193 TEST: 1.7890 0.1112 0.4081
Epoch: 0006 | TRAIN: 1.7532 0.1080 0.4197 TEST: 1.7912 0.1101 0.4074
Epoch: 0007 | TRAIN: 1.7602 0.1064 0.4174 TEST: 1.7905 0.1108 0.4050
```

Then, I changed `batch_size` from 2 to 16 and got
```
Parameter Space: ABS: 24981069.0, REL: 1.0000

LOSS FORMAT: SEMANTIC_LOSS MEAN_IOU PIX_ACC
DEPTH_LOSS ABS_ERR REL_ERR
NORMAL_LOSS MEAN MED <11.25 <22.5 <30

Epoch: 0000 | TRAIN: 2.2637 0.0390 0.2327 TEST: 2.0942 0.0687 0.3010
Epoch: 0001 | TRAIN: 1.8116 0.1151 0.3963 TEST: 1.8606 0.1208 0.3780
Epoch: 0002 | TRAIN: 1.6632 0.1466 0.4313 TEST: 1.7313 0.1421 0.4084
Epoch: 0003 | TRAIN: 1.6007 0.1554 0.4475 TEST: 1.6662 0.1462 0.4256
Epoch: 0004 | TRAIN: 1.5368 0.1669 0.4670 TEST: 1.6306 0.1563 0.4375
Epoch: 0005 | TRAIN: 1.4983 0.1667 0.4758 TEST: 1.6112 0.1499 0.4426
Epoch: 0006 | TRAIN: 1.4462 0.1733 0.4928 TEST: 1.6057 0.1569 0.4491
Epoch: 0007 | TRAIN: 1.4182 0.1808 0.4982 TEST: 1.6148 0.1499 0.4506
```

With a larger `batch_size`, the log seems to be the same as yours.

**Comment by lorenmt** [comment]
That's odd. The reason I choose a smaller batch_size simply because my GPU memory is limited. If you could increase the batch_size, please do so.

**Comment by LcDog** [comment]
Okay. Thanks for helping me check it. I really appreciate it.
